src/Features/CSharp/Portable/UseExpressionBody/Helpers/UseExpressionBodyForIndexersHelper.cs

```diff
@@ -71,5 +71,17 @@ protected override IndexerDeclarationSyntax WithBody(IndexerDeclarationSyntax de
                 declaration, options, conversionPreference,
                 out arrowExpression, out semicolonToken);
         }
+
+        protected override Location GetDiagnosticLocation(IndexerDeclarationSyntax declaration)
+        {
+            var body = GetBody(declaration);
+            if (body != null)
+            {
+                return base.GetDiagnosticLocation(declaration);
+            }
+
+            var getAccessor = GetSingleGetAccessor(declaration.AccessorList);
+            return getAccessor.ExpressionBody.GetLocation();
```

Comment:
i assume we won't even get here if the indexer only has a setter?

Reply:
Yes, that seems to the implementation for the property helper too.